src/cromshell/status/command.py

```diff
@@ -114,34 +105,50 @@ def main(config, workflow_id):
     return ret_val
 
 
-def get_metadata_status_summary(workflow_metadata):
-    """Get the status for each call in a workflow
-    and the frequency of those statuses"""
-    # workflow_metadata holds the workflow metadata as a dictionary
-    tmp_execution_status = []
-
-    # For each call in the metadata dictionary create a shortened summary containing
-    # the call name and status
-    for call in workflow_metadata["calls"]:
-        call_element = f'"{call}": '
-        execution_statuses = []
-
-        # For each call name add the number of execution_status to a list
-        for instanceDic in workflow_metadata["calls"][call]:
-            execution_statuses.append(instanceDic["executionStatus"])
-
-        # Create a key and value pair for the execution status and number of times
-        # it is seen
-        status_count = Counter(execution_statuses)
-        # For each status for the call add the call name and status name and count
-        # to list
-        for status in set(execution_statuses):
-            status_element = f'{{"{status}": {status_count[status]}}}'
-            tmp_execution_status.append(f"{call_element}{status_element}")
-
-    tmp_execution_status_json = ", ".join(tmp_execution_status)
-    tmp_execution_status_json = "{" + tmp_execution_status_json + "}"
-    return [json.loads(tmp_execution_status_json)]
+def check_for_failure(metadata: dict):
```

Comment:
How about `check_for_task_failure`?
Then the line at 114 should be removed to caller code, i.e. this function only checks if ANY of the tasks in a workflow metadata indicates failure.

Reply:
@SHuang-Broad 

The function does check for workflow failure too. If I'm not mistaken the `status` key in a dictionary indicates the status of the workflow/subworkflow and the `executionStatus` key indicates the status of a task. Line 14 will checks for a `status` key and exits if its equal to "Failed", while Line 149 does the same for `executionStatus`. My original thinking in leaving in Line 14 was if we come across a failed `status` key for workflow which sits at a higher level in the tree compared to the `executionStatus`  then we can immediately exit without having to check through all of its tasks for a failed execution. 

Line 14 can be removed if this function should only be checking the status for tasks. Should I move forward with the name change in line removal?